deployer/src/deployment/run.rs

```diff
@@ -304,10 +304,17 @@ async fn load(
     let resources = resource_manager
         .get_resources(&service_id, claim.clone())
         .await
-        .unwrap()
+        .map_err(|err| Error::Load(err.to_string()))?
         .resources
         .into_iter()
-        .map(resource::Response::from)
+        .map(|resource| {
+            resource::Response::try_from(resource).map_err(|err| Error::Load(err.to_string()))
+        })
+        // We collect into a Result so that if the response contains a resource with corrupted data,
+        // we terminate iteration and return error.
+        // TODO: investigate how the resource data can get corrupted.
+        .collect::<Result<Vec<_>>>()?
```

Comment:
This implementation errors on any corrupted resource, we could also `tracing::error!` the bad resource errors, and just return the OK ones. :thinking:

Reply:
I think returning what converted successfully is a good idea. That way the corrupted ones can be recreated and fixed.